### PR TITLE
arch/artik053: tidy up on alc5658

### PR DESCRIPTION
**Pull request by sunghan-chang**
1. Add a conditional on alc5658 header
For every header file, checking a conditional is needed to prevent
duplicated including header file. That is missed in artik053_alc5658_i2c.h.

2. refactor alc5658 files
for coding rule and readability

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: os/arch/arm/src/artik053/include/artik053_alc5658_i2c.h [hunk 1]
```diff
@@ -16,6 +16,8 @@
  *
  ****************************************************************************/
 
+#ifndef __ARCH_ARM_SRC_ARTIK05X_INCLUDE_ARTIK055_ALC5658_I2C_H
```
Comment (btheosam) [inline os/arch/arm/src/artik053/include/artik053_alc5658_i2c.h]: directory structure is arch/arm/src/artik053/include so better to keep as artik053 itself.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**